src/openforms/registrations/contrib/json_dump/plugin.py

```diff
@@ -0,0 +1,142 @@
+import base64
+
+from django.utils.translation import gettext_lazy as _
+
+from zgw_consumers.client import build_client
+
+from openforms.formio.typing import Component
+from openforms.forms.utils import to_json
+from openforms.submissions.models import (
+    Submission,
+    SubmissionFileAttachment,
+    SubmissionValueVariable,
+)
+from openforms.typing import JSONObject
+from openforms.variables.constants import FormVariableSources
+from openforms.variables.validators import validate_path_context_values
+
+from ...base import BasePlugin  # openforms.registrations.base
+from ...registry import register  # openforms.registrations.registry
+from .config import JSONDumpOptions, JSONDumpOptionsSerializer
+
+
+@register("json_dump")
+class JSONDumpRegistration(BasePlugin):
+    verbose_name = _("JSON dump registration")
+    configuration_options = JSONDumpOptionsSerializer
+
+    def register_submission(
+        self, submission: Submission, options: JSONDumpOptions
+    ) -> dict:
+        state = submission.load_submission_value_variables_state()
+
+        all_values: JSONObject = {
+            **state.get_static_data(),
+            **state.get_data(),  # dynamic values from user input
+        }
+        values = {
+            key: value
+            for key, value in all_values.items()
+            if key in options["variables"]
+        }
+
+        # Process attachments
+        self.process_variables(submission, values)
+
+        # Generate schema
+        # TODO: will be added in #4980. Hardcoded example for now.
+        schema = {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "type": "object",
+            "properties": {
+                "static_var_1": {"type": "string", "pattern": "^cool_pattern$"},
+                "form_var_1": {"type": "string"},
+                "form_var_2": {"type": "string"},
+                "attachment": {"type": "string", "contentEncoding": "base64"},
+            },
+            "required": ["static_var_1", "form_var_1", "form_var_2"],
+            "additionalProperties": False,
+        }
+
+        # Send to the service
+        json = to_json({"values": values, "schema": schema})
+        service = options["service"]
+        submission.registration_result = result = {}
```

Comment:
`submission.registration_result` automatically gets written with the return value of `register_submission`, see https://github.com/open-formulieren/open-forms/blob/794bbdae3c62eb4e4ce429661b402b57f24f4605/src/openforms/registrations/tasks.py#L338

```suggestion
```